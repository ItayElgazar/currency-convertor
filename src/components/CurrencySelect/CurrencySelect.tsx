import React, { useState, useEffect, FC, ChangeEvent } from 'react';
import {
  TextField,
  MenuItem,
  makeStyles,
  createStyles,
  Theme
} from '@material-ui/core';
import CURRENCY_LIST from '../../common/currencyList';

const useStyles = makeStyles((theme: Theme) =>
  createStyles({
    currencySelect: {
      width: 265
    }
  })
);

type CurrencySelectProps = {
  onCurrencySelected: (currency: string) => void;
  initialCurrency?: string;
  label?: string;
  disabled?: boolean;
};

const CurrencySelect: FC<CurrencySelectProps> = ({
  onCurrencySelected,
  initialCurrency,
  label,
  disabled
}) => {
  const [selectedCurrency, setSelectedCurrency] = useState(initialCurrency);

  useEffect(() => {
    setSelectedCurrency(initialCurrency);
  }, [initialCurrency]);
  const classes = useStyles();

  const handleCurrencySelect = ({ target }: ChangeEvent<HTMLInputElement>) => {
    setSelectedCurrency(target.value);
    onCurrencySelected(target.value);
  };
  return (
    <TextField
      className={classes.currencySelect}
      select
      label="Select"
      onChange={handleCurrencySelect}
      helperText={label}
      variant="outlined"
      value={selectedCurrency}
      disabled={disabled}
    >
      {CURRENCY_LIST.map(option => (
        <MenuItem key={option.code} value={option.code}>
          {option.name} - {option.code}
        </MenuItem>
      ))}
    </TextField>
  );
};

export default CurrencySelect;
