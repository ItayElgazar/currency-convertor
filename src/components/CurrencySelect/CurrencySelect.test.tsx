import React from 'react';
import renderer from 'react-test-renderer';
import CurrencySelect from './CurrencySelect';

describe('#CurerncySelect', () => {
  it('renders correctly', () => {
    const currencySelect = renderer
      .create(<CurrencySelect onCurrencySelected={jest.fn()} />)
      .toJSON();
    expect(currencySelect).toMatchSnapshot();
  });
});
