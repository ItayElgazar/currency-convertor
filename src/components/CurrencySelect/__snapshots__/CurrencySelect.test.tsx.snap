// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`#CurerncySelect renders correctly 1`] = `
<div
  className="MuiFormControl-root MuiTextField-root makeStyles-currencySelect-1"
>
  <label
    className="MuiFormLabel-root MuiInputLabel-root MuiInputLabel-formControl MuiInputLabel-animated MuiInputLabel-outlined"
    data-shrink={false}
  >
    Select
  </label>
  <div
    className="MuiInputBase-root MuiOutlinedInput-root MuiInputBase-formControl"
    onClick={[Function]}
  >
    <div
      aria-haspopup="listbox"
      aria-labelledby=" "
      className="MuiSelect-root MuiSelect-select MuiSelect-selectMenu MuiSelect-outlined MuiInputBase-input MuiOutlinedInput-input"
      onBlur={[Function]}
      onFocus={[Function]}
      onKeyDown={[Function]}
      onMouseDown={[Function]}
      role="button"
      tabIndex={0}
    >
      <span
        dangerouslySetInnerHTML={
          Object {
            "__html": "&#8203;",
          }
        }
      />
    </div>
    <input
      autoFocus={false}
      onAnimationStart={[Function]}
      type="hidden"
    />
    <svg
      aria-hidden="true"
      className="MuiSvgIcon-root MuiSelect-icon MuiSelect-iconOutlined"
      focusable="false"
      role="presentation"
      viewBox="0 0 24 24"
    >
      <path
        d="M7 10l5 5 5-5z"
      />
    </svg>
    <fieldset
      aria-hidden={true}
      className="PrivateNotchedOutline-root-83 MuiOutlinedInput-notchedOutline"
    >
      <legend
        className="PrivateNotchedOutline-legendLabelled-85"
      >
        <span>
          Select
        </span>
      </legend>
    </fieldset>
  </div>
</div>
`;
